Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -395,25 +403,38 @@ def fetch_notables(service: client.Service, mapper: UserMappingObject, comment_t
     # we didn't get any new incident or get less then limit
     # so the next run earliest time will be the latest_time from this iteration
     # should also set when num_of_dropped == FETCH_LIMIT
-    if not incidents or (len(incidents) + num_of_dropped) < FETCH_LIMIT:
+    if (len(incidents) + num_of_dropped) < FETCH_LIMIT:
+        demisto.debug(f'[SplunkPy] Number of fetched incidents, dropped or not, is less than {FETCH_LIMIT=}. Starting new page')
         next_run_earliest_time = latest_time
         new_last_run = {
             'time': next_run_earliest_time,
             'latest_time': None,
             'offset': 0,
             'found_incidents_ids': last_run_fetched_ids
         }
-
     # we get limit notables from splunk
     # we should fetch the entire queue with offset - so set the offset, time and latest_time for the next run
     else:
+        demisto.debug(f'[SplunkPy] Number of fetched incidents, dropped or not, is equal to {FETCH_LIMIT=}. Continue pagination')
         new_last_run = {
             'time': occured_start_time,
             'latest_time': latest_time,
             'offset': search_offset + FETCH_LIMIT,
             'found_incidents_ids': last_run_fetched_ids
         }
-    demisto.debug(f'SplunkPy - {new_last_run["time"]=}, {new_last_run["latest_time"]=}, {new_last_run["offset"]=}')
+    new_last_run['late_indexed_pagination'] = False
+    # Need to fetch again this "window" to be sure no "late" indexed events are missed
+    if num_of_dropped == FETCH_LIMIT and '`notable`' in fetch_query:
```

Comment:
i'm wondering from the back&forth of the pagination
in case of 50 dropped:
- late_indexed_pagination
- 0 lat incidents, return to regular fetch
-- 50 dropped, late_indexed_pagination (due to we are on the near time because of the `look back`)
-- 0 lat incidents, return to regular fetch
---- 50 dropped, late_indexed_pagination (due to we are on the near time because of the `look back`)
---- 0 lat incidents, return to regular fetch

-----------
until we go to a new 'window' which excluded the index time.

may be we can store in the last run, counter (instead of just `True`) about how much time we do the `late_indexed_pagination` and stop it after, let say, 4 times?

Reply:
We discussed this offline and we saw that this isn't an issue.